### PR TITLE
Flatsat test scripts initial PR

### PR DESCRIPTION
**Pull request by jlazzz**
Flatsat test scripts are not complete, but figure it's an appropriate time to merge this branch since **the architecture is all there** and I don't want this branch to live even longer than it already has. All the changes are within the /test directory so shouldn't affect any other work.

### CODE REVIEW

File: test/functionalTestLib.py [hunk 1]
```diff
@@ -0,0 +1,1019 @@
...
+        testPassed = "Pass"
+        # 1) Perform the control computer health check
+        # 1a) Ensure that the OBC is turned on and has the most up-to-date firmware installed (Note this version down somewhere)
+
+        # 1b) Power on the CubeADCS and wait one second for everything to power on
+        input("Please power on the CubeADCS. Press enter to continue.")
+        time.sleep(1)
+
+        # 1c) Verify results in Table 2 from the Full and Partial Functional Test Plan
+        server, port, toSend = gs.getInput("obc.adcs.adcs_get_bootloader_state")
```
Comment (Taylorr82) [inline test/functionalTestLib.py:63]: use of 'obc' instead of a satellite

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: test/test_flatsat_dfgm.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+test = test() #call to initialize local test class
+
+# TODO - Automate the remaining steps in the DFGM Data Acquisition and S Band Downlink test - 3, 4
+def test_DFGM_dataAcquisitionAndSBandDownlink():
+    testPassed = "Pass"
+    # 1) Ensure that the OBC, UHF, and EPS are turned on, and that the OBC has the most up-to-date firmware installed (Doesn't have to be automated)
+
+    # 2) Turn the DFGM on for 10 seconds and have the OBC process the data output by the DFGM into 1 Hz, 10 Hz, and 100 Hz data. Store them as separate
+    # files on the SD card. This data will just be used for testing, and will consist of background magnetic signatures
+    server, port, toSend = gs.getInput('obc.dfgm.dfgm_run(10)')
```
Comment (Taylorr82) [inline test/test_flatsat_dfgm.py:45]: Missed some OBCs

**Review by Taylorr82 — CHANGES_REQUESTED** [review]
Perhaps the tests should have a way of knowing which satellite they're talking to

**Comment by joshdellaz** [comment]
Perhaps that will be in a new feature branch in the future.

**Comment by Taylorr82** [comment]
The tests won't pass with this in the current state with 'obc' everywhere, and the only obvious way to fix it is to pass the satellite somehow. Can't merge non passing tests

**Comment by joshdellaz** [comment]
I added a new commit fixing all the obc. To ex2.

**Comment by Taylorr82** [comment]
https://github.com/AlbertaSat/ex2_ground_station_software/pull/115#discussion_r919555455

**Comment by joshdellaz** [comment]
Poke

File: test/testLib.py [hunk 3]
```diff
@@ -12,76 +12,750 @@
...
- * @file commandParser.py
- * @author Dustin Wagner
+ * @file testLib.py
+ * @author Dustin Wagner, Daniel Sacro
  * @date 2021-06-28
 '''
 
-'''This File contains no actual tests. Only a class to send/receive and format results of tests'''
+'''In addition to sending/receiving CSP packets and formatting test results, the class in this 
+file also contains a function that runs the OBC system-wide housekeeping test'''
```
Comment (Taylorr82) [inline test/testLib.py:21]: This is too much responsibility for this class. It needs to only send/receive and format. It is not a test, it's a facillitator

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: test/testLib.py (hunk 3 above)
Comment (joshdellaz) [reply test/testLib.py:21]: I'm not sure what you're suggesting. Could you elaborate?

Comment (joshdellaz) [reply test/testLib.py:21]: Poke

Comment (Taylorr82) [reply test/testLib.py:21]: This is covered by my task to refactor the tests

Comment (joshdellaz) [reply test/testLib.py:21]: @tganley

**Comment by Taylorr82** [comment]
Replaced by https://github.com/AlbertaSat/ex2_ground_station_software/pull/146

**Pull request closed without merging**